### ISSUE TITLE
Producing shared libraries

### ISSUE THREAD

**Issue by rouson** [fortran-lang/fpm#681]
### Description

Is anyone working on enabling `fpm` to build shared libraries?  I would like to offer `fpm` as an alternative way to build [OpenCoarrays].  In thinking through what `fpm` won't be able to match relative to the current `cmake` build system generator, an ability to produce shared libraries seems like a useful one at least for system administrators.  This feature request overlaps some with issue  #524.

[OpenCoarrays]: https://github.com/sourceryinstitute/opencoarrays

### Possible Solution

I don't know enough to suggest possible solutions.

**Comment by awvwgk** [comment]
No, this is currently not possible. See my answer in https://github.com/fortran-lang/fpm/discussions/655 for things to consider in case anyone wants to pick this up.

**Comment by wyphan** [comment]
Thanks @awvwgk for mentioning this in the Spack issue #79.

Spack _does_ handle `RPATH` issues quite nicely for Linux and MacOS: https://spack.readthedocs.io/en/latest/packaging_guide.html#handling-rpaths
I'm not too familiar with DLLs to comment on how to resolve this for Windows.

**Comment by ivan-pi** [comment]
On *nix platforms at least, it may be possible to use [libtool](https://www.gnu.org/software/libtool/) as an intermediary.
